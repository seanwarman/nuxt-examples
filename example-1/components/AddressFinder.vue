<template>
  <b-form class="mb-5" @submit="(e) => e.preventDefault()">
    <p class="mb-2">Enter a postcode</p>
    <div class="mb-3 d-flex flex-wrap">
      <b-form-input
        :id="`postCodeInput${index === null ? '' : index}${
          type === null ? '' : type
        }`"
        v-model="postCodeInput"
        :disabled="disabled"
        class="post-code-input mr-3 mb-3"
        type="text"
        placeholder="Postcode"
        @keyup.enter="findAddress"
      />
      <b-button
        :variant="
          postCodeInput.length ? 'outline-primary' : 'outline-secondary'
        "
        :disabled="disabled"
        class="mb-3"
        @click="findAddress"
      >
        Find address
      </b-button>
    </div>
    <template v-if="showNoAddressesFoundError">
      <p class="error">No addresses found</p>
    </template>
    <template v-if="addressDropDownVisible">
      <p class="font-weight-bold">Select an address from the list</p>
      <b-dropdown
        :text="`${addressOptions.length} address${
          addressOptions.length === 1 ? '' : 'es'
        } found`"
        :disabled="disabled"
        variant="outline-primary"
        class="align-self-start mb-3"
      >
        <b-dropdown-item
          v-for="(option, i) in addressOptions"
          :key="i"
          @click="setAddress(option)"
        >
          {{ option.address[0]
          }}{{ option.address[1] && ", " + option.address[1] }}
        </b-dropdown-item>
      </b-dropdown>
    </template>
    <p
      v-if="!enterFullAddress"
      class="link mb-3"
      @click="!disabled && (enterFullAddress = true)"
    >
      I can't enter a UK postcode
    </p>
    <template
      v-if="
        enterFullAddress ||
        address1.length ||
        address2.length ||
        address3.length ||
        city.length ||
        postCode.length
      "
    >
      <p class="mb-1">Building and street</p>
      <div>
        <b-form-invalid-feedback
          id="address1-feedback"
          :state="validateState('address1')"
          class="error mb-1"
        >
          Invalid
        </b-form-invalid-feedback>
        <b-form-input
          :id="`address1${index === null ? '' : index}${
            type === null ? '' : type
          }`"
          :ref="`address1${index === null ? '' : index}${
            type === null ? '' : type
          }`"
          v-model="$v.address1.$model"
          :disabled="disabled"
          :state="validateState('address1')"
          class="form-input mb-1"
          type="text"
        />
      </div>
      <b-form-input
        v-model="address2"
        :disabled="disabled"
        class="form-input mb-1"
        type="text"
      />
      <b-form-input
        v-model="address3"
        :disabled="disabled"
        class="form-input mb-1"
        type="text"
      />
      <p class="mb-1">Town or city</p>
      <div>
        <b-form-invalid-feedback
          id="city-feedback"
          :state="validateState('city')"
          class="error mb-1"
        >
          Invalid
        </b-form-invalid-feedback>
        <b-form-input
          :id="`city${index === null ? '' : index}${type === null ? '' : type}`"
          :ref="`city${index === null ? '' : index}${
            type === null ? '' : type
          }`"
          v-model="$v.city.$model"
          :disabled="disabled"
          :state="validateState('city')"
          class="form-input mb-1"
          type="text"
        />
      </div>
      <p class="mb-1">Postcode</p>
      <div class="mb-5">
        <b-form-invalid-feedback
          id="postCode-feedback"
          :state="validateState('postCode')"
          class="error mb-1"
        >
          Invalid
        </b-form-invalid-feedback>
        <b-form-input
          :id="`postCode${index === null ? '' : index}${
            type === null ? '' : type
          }`"
          :ref="`postCode${index === null ? '' : index}${
            type === null ? '' : type
          }`"
          v-model="$v.postCode.$model"
          :disabled="disabled"
          :state="validateState('postCode')"
          class="form-input mb-1"
          type="text"
        />
      </div>
    </template>
  </b-form>
</template>

<script>
import Vue from "vue";
import Vuelidate from "vuelidate";
import { required, maxLength } from "vuelidate/lib/validators";
import { isPostcode } from "~/lib/validation";
import { addressLookupHeaders } from "~/static/addressLookupConstants.js";
Vue.use(Vuelidate);
export default {
  props: {
    index: {
      type: Number,
      default: null,
    },
    type: {
      type: String,
      default: null,
    },
    address: {
      type: Object,
      default() {
        return {
          address1: String,
          address2: String,
          address3: String,
          city: String,
          postCode: String,
        };
      },
    },
    disabled: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      postCodeInput: "",
      addressDropDownVisible: false,
      addressOptionsVisible: false,
      showNoAddressesFoundError: false,
      addressOptions: [],
      address1: "",
      address2: "",
      address3: "",
      city: "",
      postCode: "",
      enterFullAddress: false,
    };
  },
  mounted() {
    if (
      this.address.address1 ||
      this.address.address2 ||
      this.address.address3 ||
      this.address.city ||
      this.address.postCode
    ) {
      this.mapPropsToState();
      this.enterFullAddress = true;
    }
  },
  methods: {
    mapPropsToState() {
      this.address1 = this.address.address1;
      this.address2 = this.address.address2;
      this.address3 = this.address.address3;
      this.city = this.address.city;
      this.postCode = this.address.postCode;
      this.postCodeInput = "";
    },
    findAddress() {
      if (this.postCodeInput.length) {
        this.$axios
          .$get(
            `https://api.addressian.co.uk/v2/autocomplete/${this.postCodeInput}`,
            addressLookupHeaders
          )
          .then((response) => {
            response.sort(
              (a, b) => parseFloat(a.address) - parseFloat(b.address)
            );
            this.addressOptions = response;
            if (response.length) {
              this.showNoAddressesFoundError = false;
              this.addressDropDownVisible = true;
            } else {
              this.showNoAddressesFoundError = true;
              this.addressDropDownVisible = false;
            }
          })
          .catch(() => {
            this.showNoAddressesFoundError = true;
            this.addressDropDownVisible = false;
          });
      }
    },
    setAddress(option) {
      this.address1 = option.address[0];
      this.address2 = option.address[1];
      this.address3 = option.address[2];
      this.city = option.city;
      this.postCode = option.postcode;
      this.addressOptionsVisible = false;
      this.enterFullAddress = true;
    },
    validateState(name) {
      const { $dirty, $error } = this.$v[name];
      return $dirty ? !$error : null;
    },
  },
  validations: {
    address1: {
      required,
      maxLength: maxLength(100),
    },
    address2: {
      maxLength: maxLength(100),
    },
    address3: {
      maxLength: maxLength(100),
    },
    city: {
      required,
      maxLength: maxLength(100),
    },
    postCode: {
      required,
      isPostcode,
      maxLength: maxLength(100),
    },
  },
};
</script>

<style lang="scss">
.post-code-input {
  width: 140px;
}
.link {
  color: $dbw-green;
  &:hover {
    cursor: pointer;
  }
}
.error {
  font-size: 80%;
  color: #dc003e;
}
</style>
