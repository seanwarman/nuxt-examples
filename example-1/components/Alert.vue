<template>
  <div :class="`dbw-alert ${variant ? variant : ''} d-flex flex-row mb-3`">
    <img
      v-if="variant"
      class="dbw-alert-icon mr-3"
      :src="
        displayIcon
          ? require(`~/assets/icons/${
              variant === 'danger' || variant === 'warning'
                ? 'exclamation-triangle-solid.svg'
                : variant === 'success'
                ? 'check-circle-solid.svg'
                : (displayIcon = false)
            }`)
          : ''
      "
    />
    <div class="d-block">
      <slot></slot>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    variant: {
      type: String,
      default: null,
    },
  },
  data() {
    return {
      displayIcon: true,
    };
  },
};
</script>

<style lang="scss">
.dbw-alert {
  padding: 16px;
  border-radius: 2px;
  font-weight: 800;
  &.warning {
    background-color: $dbw-yellow;
    color: $dbw-black;
    .dbw-alert-icon {
      filter: invert(0.2);
    }
  }
  &.success {
    background-color: $dbw-green;
    color: $dbw-white;
    .dbw-alert-icon {
      filter: invert(1);
    }
  }
  &.danger {
    background: $dbw-red;
    color: $dbw-white;
    .dbw-alert-icon {
      filter: invert(1);
    }
  }
}
.dbw-alert-icon {
  width: 24px;
  height: 24px;
  flex-shrink: 0;
}
</style>
