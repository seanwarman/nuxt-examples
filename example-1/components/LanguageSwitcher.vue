<template>
  <b-button-group size="sm">
    <b-button
      :variant="$i18n.locale === 'en' ? selectedVariant : unselectedVariant"
      @click="switchLanguage('en')"
      >En 🏴󠁧󠁢󠁥󠁮󠁧󠁿</b-button
    >
    <b-button
      :variant="$i18n.locale === 'cy' ? selectedVariant : unselectedVariant"
      @click="switchLanguage('cy')"
      >Cy 🏴󠁧󠁢󠁷󠁬󠁳󠁿</b-button
    >
  </b-button-group>
</template>

<script>
export default {
  props: {
    selectedVariant: { type: String, default: "light" },
    unselectedVariant: { type: String, default: "outline-light" },
  },
  methods: {
    switchLanguage(languageCode) {
      this.$i18n.setLocale(languageCode);
      this.$store.commit("Global/setLanguage", languageCode);
      document.documentElement.setAttribute("lang", languageCode);
    },
  },
};
</script>
