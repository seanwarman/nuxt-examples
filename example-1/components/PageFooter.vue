<template>
  <div>
    <div
      v-if="$route.path.includes('/user-account-area/')"
      class="faq-modal-footer"
    >
      <FAQModal />
    </div>
    <div class="footer">
      <b-container class="d-flex justify-content-end align-items-center h-100">
        <img
          class="footer-logo"
          src="~/assets/icons/dev-bank-wales-logo-white.png"
        />
      </b-container>
    </div>
  </div>
</template>

<script>
import FAQModal from "~/components/user-account-area/FAQModal.vue";
export default {
  components: {
    FAQModal,
  },
};
</script>

<style lang="scss">
.footer {
  height: 88px;
  background-color: #3c3c3c;
}
.footer-logo {
  width: 120px;
}
.faq-modal-footer {
  padding: 20px;
  background-color: $dbw-background-green;
  display: flex;
  align-items: center;
  justify-content: center;
  @media screen and (min-width: map-get($grid-breakpoints, "lg")) {
    display: none;
  }
}
</style>
