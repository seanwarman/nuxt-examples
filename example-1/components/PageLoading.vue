<template>
  <div class="wrapper d-flex align-items-center justify-content-center">
    <img class="loading-wheel" src="/loadingwheel3.gif" />
  </div>
</template>

<script></script>

<style lang="scss">
.wrapper {
  width: 100%;
  height: 100vh;
}
.loading-wheel {
  width: 40px;
  height: 40px;
  filter: hue-rotate(240deg) brightness(1.1);
}
</style>
