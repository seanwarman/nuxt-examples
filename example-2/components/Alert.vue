<template>
  <b-alert :variant="variant" :show="show"><slot /></b-alert>
</template>

<script>
export default {
  props: {
    show: {
      type: Boolean,
      default: false,
    },
    variant: {
      type: String,
      default: 'info',
    },
  },
}
</script>

<style lang="scss" scoped>
.alert-info {
  color: $alert-info-color;
  background-color: $alert-info-bg-color;
  border-color: $alert-info-border-color;
}

.alert ::v-deep a,
.alert ::v-deep a:link,
.alert ::v-deep a:visited {
  color: currentColor;
  text-decoration: underline;
}

.alert ::v-deep a {
  &:focus,
  &:focus-visible {
    background: $brand-primary;
    color: #ffffff;
    box-shadow: 0 0 0 4px $brand-primary;
    outline: none;
  }
}
</style>
