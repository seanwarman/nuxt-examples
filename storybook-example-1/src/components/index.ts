export { default as TzButton } from './Action/Tz-Button/Tz-Button.vue';
export { default as TzCheckoutCta } from './Action/Tz-Checkout-CTA/Tz-Checkout-Cta.vue';
export { default as TzModalDialog } from './Action/Tz-ModalDialog/Tz-ModalDialog.vue';
export { default as TzPayPalButton } from './Action/Tz-Button/Tz-PayPalButton.vue';
export { default as TzProgressBar } from './Action/Tz-Progress-Bar/Tz-Progress-Bar.vue';

export { default as TzBasketAccordion } from './Informational/Tz-Basket-Accordion/Tz-Basket-Accordion.vue';
export { default as TzBasketCard } from './Informational/Tz-Basket-Card/Tz-Basket-Card.vue';
export { default as TzBasketTotalsSummary } from './Informational/Tz-Basket-Totals-Summary/Tz-Basket-Totals-Summary.vue';
export { default as TzBundleCreator } from './Informational/Tz-BundleCreator/Tz-BundleCreator.vue';
export { default as TzC2wBanner } from './Informational/Tz-C2W-Banner/Tz-C2W-Banner.vue';
export { default as TzExpressCheckout } from './Informational/Tz-ExpressCheckout/Tz-ExpressCheckout.vue';
export { default as TzInfoBox } from './Informational/Tz-InfoBox/Tz-InfoBox.vue';
export { default as TzJumboCard } from './Informational/Tz-Jumbo-Card/Tz-Jumbo-Card.vue';
export { default as TzOrderItemsSummary } from './Informational/Tz-Order-Items-Summary/Tz-Order-Items-Summary.vue';
export { default as TzOrderTotalsSummary } from './Informational/Tz-Order-Totals-Summary/Tz-Order-Totals-Summary.vue';
export { default as TzProductTile } from './Informational/Tz-Product-Tile/Tz-Product-Tile.vue';
export { default as TzUspRibbon } from './Informational/Tz-USP-Ribbon/Tz-USP-Ribbon.vue';

export { default as TzInputField } from './Input/Tz-InputField/Tz-InputField.vue';
export { default as TzLogin } from './Input/Form/Tz-Login/Tz-Login.vue';
export { default as TzPaymentMethods } from './Input/Tz-PaymentMethods/Tz-PaymentMethods.vue';
export { default as TzRadioField } from './Input/Tz-RadioField/Tz-RadioField.vue';
export { default as TzSelectField } from './Input/Tz-SelectField/Tz-SelectField.vue';

export { default as TzNavigation } from './Navigation/Tz-Navigation/Tz-Navigation.vue';

export { default as TzContainer } from './Structural/Tz-Container/Tz-Container.vue';

export { default as TzAccountIcon } from './Informational/Icons/Tz-Account-Icon.vue';
export { default as TzArrowIcon } from './Informational/Icons/Tz-Arrow-Icon.vue';
export { default as TzBasketIcon } from './Informational/Icons/Tz-Basket-Icon.vue';
export { default as TzCaretIcon } from './Informational/Icons/Tz-Caret-Icon.vue';
export { default as TzCheckmarkIcon } from './Informational/Icons/Tz-Checkmark-Icon.vue';
export { default as TzChevronIcon } from './Informational/Icons/Tz-Chevron-Icon.vue';
export { default as TzCloseIcon } from './Informational/Icons/Tz-Close-Icon.vue';
export { default as TzExclaimationIcon } from './Informational/Icons/Tz-Exclaimation-Icon.vue';
export { default as TzExpandIcon } from './Informational/Icons/Tz-Expand-Icon.vue';
export { default as TzFilterIcon } from './Informational/Icons/Tz-Filter-Icon.vue';
export { default as TzSearchIcon } from './Informational/Icons/Tz-Search-Icon.vue';
export { default as TzWishlistIcon } from './Informational/Icons/Tz-Wishlist-Icon.vue';
