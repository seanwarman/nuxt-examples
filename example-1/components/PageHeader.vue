<template>
  <div>
    <div class="header-top">
      <b-container class="d-flex align-items-center h-100">
        <img
          class="header-top-logo"
          src="~/assets/icons/welsh-gov-initiative-logo.png"
        />
        <language-switcher class="ml-auto" />
      </b-container>
    </div>
    <div class="header-bottom">
      <b-container class="d-flex flex-column justify-content-start h-100">
        <img
          v-if="!loading"
          class="header-bottom-logo"
          :src="require(`~/assets/icons/${LogoPath}`)"
          @click="goHome"
        />
        <div class="divider-line w-100" />
      </b-container>
    </div>
  </div>
</template>

<script>
import LanguageSwitcher from "./LanguageSwitcher.vue";
export default {
  components: { LanguageSwitcher },
  data() {
    return {
      loading: true,
    };
  },
  computed: {
    Brand: {
      get() {
        return this.$store.getters["Global/Brand"];
      },
    },
    LogoPath: {
      get() {
        if (this.Brand === "sbw") return "self-build-wales-logo.svg";
        else return "dev-bank-wales-logo.png";
      },
    },
  },
  mounted() {
    this.loading = false;
  },
  methods: {
    goHome() {
      $nuxt.$emit("takeUserToDefaultHomeScreen");
      this.$store.commit("Global/resetSelectedCase");
      this.$router.push("/user-account-area/my-applications");
    },
  },
};
</script>

<style lang="scss">
.header-top {
  height: 50px;
  background-color: #3c3c3c;
}
.header-top-logo {
  height: 30px;
}
.header-bottom-logo {
  height: 64px;
  width: 147px;
  margin: 25px 0 20px 0;
  &:hover {
    cursor: pointer;
  }
  @media screen and (max-width: map-get($grid-breakpoints, "lg")) {
    height: 53px;
    width: 122px;
    margin: 16px 0;
  }
}
.divider-line {
  height: 1px;
  background-color: #3c3c3c;
  opacity: 0.1;
}
</style>
