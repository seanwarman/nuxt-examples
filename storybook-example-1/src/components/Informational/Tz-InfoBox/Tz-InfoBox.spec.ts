/**
 * @vitest-environment jsdom
 */

import { VueWrapper, mount } from '@vue/test-utils';
import { expect, describe, beforeEach, it } from 'vitest';
import { TzInfoBox } from '@/components';

let wrapper: VueWrapper;

describe('TzInfoBox', () => {
  beforeEach(() => {
    wrapper = mount(TzInfoBox, {
      props: {
        actionLabel: 'Edit'
      }
    });
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('when the action is clicked', () => {
    it("will emit an 'actionClick' event", () => {
      wrapper.find('button').trigger('click');
      expect(wrapper.emitted()).toHaveProperty('actionClick');
    });
  });
});
