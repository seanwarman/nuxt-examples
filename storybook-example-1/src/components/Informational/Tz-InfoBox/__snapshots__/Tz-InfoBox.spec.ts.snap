// Vitest Snapshot v1

exports[`TzInfoBox > renders correctly 1`] = `
<div
  class="flex flex-col gap-1.5"
>
  <!--v-if-->
  <div
    class="relative flex flex-grow flex-col gap-0 border p-4"
  >
    <div
      class="flex-grow"
    >
      
      
    </div>
    <div
      class="mt-3 text-right"
    >
      <button
        class="whitespace-nowrap rounded-sm bg-gray-100 py-2 px-3.5 text-xs font-bold uppercase leading-4 tracking-widest hover:underline focus:underline sm:px-7"
      >
        Edit
      </button>
    </div>
  </div>
</div>
`;
